### PR TITLE
Data entity document improvement

### PR DESCRIPTION
**Pull request by ajithkumar-maragathavel**
### Description

Improved the Data entity MFTF document by providing examples for each data entity node.
Added example from the already defined MFTF test for each sub-topic.

### Affected Pages
- https://devdocs.magento.com/mftf/docs/data.html

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/verification tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)
 - [x] Changes to Framework doesn't have backward incompatible changes for tests or have related Pull Request with fixes to tests

### CODE REVIEW

File: docs/data.md [hunk 1]
```diff
@@ -32,6 +48,12 @@ In this example:
 *  `MAGENTO_ADMIN_USERNAME` is a name of an environment variable.
    The corresponding value will be assigned to `userInput` as a result.
 
+The following is the usage of `_ENV` in the `Magento/Braintree/Test/Mftf/ActionGroup/AdminDeleteRoleActionGroup.xml` action group:
```
Comment (dobooth) [inline docs/data.md:51]: "The following is an example of the usage of..." 
Same with L72 and L96.

File: docs/data.md [hunk 2]
```diff
@@ -107,7 +145,7 @@ userInput="We'll email you an order confirmation with details and tracking info.
 
 ## Format
 
-The format of `<data>` is:
+The format of `<data>` entity is:
```
Comment (dobooth) [inline docs/data.md:148]: "The format of the `data`...

File: docs/data.md [hunk 3]
```diff
@@ -205,7 +243,7 @@ You can also call data from the xml definition of a `data` tag directly:
 
 Attributes|Type|Use|Description
 ---|---|---|---
-`name`|string|optional|Name of the `<entity>`.
+`name`|string|optional|Name of the `<entity>`. Camel case is used for the entity name.
```
Comment (dobooth) [inline docs/data.md:246]: "Use camel case for entity names."

File: docs/data.md [hunk 4]
```diff
@@ -231,6 +275,12 @@ Attributes|Type|Use|Description
 `entityKey`|string|optional|Key attribute of the referenced entity from which to get a value.
 `unique`|--|--|*This attribute hasn't been implemented yet.*
 
+Example:
+
+```
```
Comment (dobooth) [inline docs/data.md:280]: Use xml as the codeblock type.

File: docs/data.md (hunk 1 above)
Comment (ajithkumar-maragathavel) [reply docs/data.md]: @dobooth Updated the changes.

File: docs/data.md (hunk 2 above)
Comment (ajithkumar-maragathavel) [reply docs/data.md]: @dobooth Updated the changes.

File: docs/data.md (hunk 3 above)
Comment (ajithkumar-maragathavel) [reply docs/data.md]: @dobooth Updated the changes. Please review it.

File: docs/data.md (hunk 4 above)
Comment (ajithkumar-maragathavel) [reply docs/data.md]: @dobooth This has been done. Please review it.

**Pull request merged**